storyStarterGame.js

```diff
@@ -0,0 +1,470 @@
+//     let afterLife = readLineSync.keyInSelect (arr1)
+//    let yourchoice = arr1[afterLife] 
+//     if (yourchoice === "avenge death") {
+//     avengeDeath ();
+// } else if (yourchoice === "haunt criminals") {
+//     hauntCriminals();
+// } else if (yourchoice === "rest peacefully") {
+//     restPeacefully();
+// } else {
+//     leaveGame();
+// }
+// } 
+let i =0
+let i2=1
+arr1 = ["avenge death", "haunt criminals", "rest peacefully"]
+if (arr1.length===1) {
+    if (arr1[i]==="avenge death"){
+        let afterLife =readLineSync.keyInSelect(arr1, "Do you want to avenge their death? ", {cancel:"Must choose above option"})
+        if (arr1[afterLife] === "avenge death") {
+            arr1.splice(0,1)
+            avengeDeath()
+        }
+    }else if (arr1[i] === "haunt criminals") {
+        let afterLife = readLineSync.keyInSelect(arr1, "Do you want to haunt criminals? ", {cancel:"Must choose above option"})
+        if (arr1[afterLife]=== "haunt criminals") {
+            arr1.splice(0,1)
+            hauntCriminals()
+        }
+    }else if (arr1[i] === "rest peacefully") {
+        let afterLife=readLineSync.keyInSelect(arr1, "Do you want them to rest peacefully?", {cancel: "Must choose above option"})
+        if (arr1[afterLife] === "rest peacefully") {
+            arr1.splice(0,1)
+            restPeacefully()
+        }
+    }
+} else if (arr1.length===2){
+     if (arr1[i] === "avenge death" && arr1[i2] === "haunt criminals"){
+        let afterLife = readLineSync.keyInSelect(arr1, "Do you want them to avenge their death or haunt criminals ? ", {cancel: "Must choose above options"})
+         if (arr1[afterLife]==="avenge death") {
+            arr1.splice(afterLife,1) 
+            avengeDeath()
+         }else if (arr1[afterLife] === "haunt criminals") {
+            arr1.splice(afterLife,1)
+            hauntCriminals()
+         }
+     } else if (arr1[i] === "rest peacefully" && arr1[i2] === "haunt criminals") {
+         let afterLife = readLineSync.keyInSelect(arr1, "Do you want them to haunt criminals or rest peacefully? ", {cancel: "Must choose above options"})
+         if (arr1[afterLife] === "haunt criminals") {
+             arr1.splice(afterLife,1)
+             hauntCriminals()
+         } else if (arr1[afterLife] === "rest peacefully") {
+             arr1.splice(afterLife,1)
+             restPeacefully()
+         }
+     } else if (arr1[i] === "avenge death" && arr1[i2] === "rest peacefully") {
+         let afterLife = readLineSync.keyInSelect(arr1, "Do you want them to avenge their death or rest peacefully? ", {cancel: "Must choose above options"})
+         if (arr1[afterLife] === "rest peacefully") {
+             arr1.splice(afterLife,1)
+             restPeacefully()
+         } else if (arr1[afterLife] === "avenge death") {
+             arr1.splice(afterLife, 1)
+             avengeDeath()
+         }
+     }
+} else if (arr1.length=== 0) {
+    replayTheGame()
+}
+}
```

Comment:
`arr` here is never defined and its assigned to a array of length `3` so its never going to meet none of these if/else conditions